boards/README.md

```diff
@@ -6,6 +6,7 @@ that Tock supports.
 
 | Board                                                                | Architecture    | MCU            | Interface  | App deployment | QEMU Support? |
 |----------------------------------------------------------------------|-----------------|----------------|------------|----------------|---------------|
+| [Raspberry Pi Pico](raspberry_pi_pico/README.md)                     | ARM Cortex-M0+  | RP2040         | openocd    | openocd        | No            |
```

Comment:
I noticed that the boards were ordered based on the alphabetical order of the chip. If this is different, I can move it to the bottom.

Reply:
@bradjc should this be placed in alphabetical order or at the bottom?

Reply:
I think you mean by alphabetical order of the arch? I think that the boards are grouped by arch, but the order ending up being alphabetical is mostly just luck because there are really only 3 architectures so far. I think putting this at the end of the ARM boards makes sense

Reply:
Done :)